pallets/block-rewards/src/mock.rs

```diff
@@ -161,7 +153,7 @@ orml_traits::parameter_type_with_key! {
 }
 
 impl orml_tokens::Config for Test {
-	type Amount = i64;
+	type Amount = IBalance;
```

Comment:
I would say using here `i64` is preferible (or `type IBalanace = i64` on top of this file) instead of using `cfg_primitives::IBalance`. Actually, `cfg_primitive::IBalance` should not be a "primitive", it's our centrifuge domain type used to configure our runtimes, so it is at the runtime level. From pallets, we should not use "runtime things". These types should probably be moved to some runtime common crate in the future.

Currently it's a mess. And primitives/utilities/and types are quite mixed. No CR here. Just a comment!

Reply:
Unfortunately, using `i64` leads to overflows! Will use `i128` then.

Reply:
Oh, that's weird. I guess it's due to the use of amounts with a lot of decimals, right?

Reply:
Yes, if I use CFG denomination, then it quickly creates an overflow